### ISSUE TITLE
Bug: Obs disappear with filtered view

### ISSUE THREAD

**Issue by kbailey-noaa** [asascience-open/IOOS-EDS#188]
@Bobfrat 
If I turn on CO-OPS water level stations layer and then in the filter bar type in "SFBOFS" (or any key word EXCEPT the words in the CO-OPS water level layer name (e.g. water)) then the CO-OPS markers disappear from the map, even though the layer is technically still on.  If I clear the filter field to back to the full TOC, then the CO-OPS obs markers reappear.

**Comment by kknee** [comment]
@kbailey-noaa this isn't a bug, it's how the filters were setup to work. The filter filters through the stations and identifies the ones that match your key word.  So if one of the stations also had SBOFS in the name or parameter list, it would show up.

**Comment by kbailey-noaa** [comment]
The active layer’s markers are intentionally removed from the map when filtering in the TOC list? I understand the layer would disappear from the TOC list but it doesn’t make sense that the markers would also disappear from the map. 

We can table til after Thursday. Im just trying to capture things I find as I use the Viewer.

**Comment by kknee** [comment]
Yes, they are intentionally removed.  The current approach allows users to search within an obs layer for specific parameters, e.g. if you search for 'velocity' it will show only the stations with parameters that include that keyword.

What you are seeing is the downside to this approach....

I agree, this merits a larger discussion as being able to search w/in an obs layer is useful

**Comment by Bobfrat** [comment]
I think it may be a bug. If a model layer is on and you use the filter bar it returns the results that match and keeps your model layer on.

![Screen Shot 2019-08-07 at 10 58 40 AM](https://user-images.githubusercontent.com/5702672/62633571-5e9e0c00-b902-11e9-858a-00e50322db46.png)

**Comment by kknee** [comment]
@Bobfrat  - right, for obs it keeps the layer in too, but doesn't draw any of the markers if the search term doesn't match up with any of the parameter names.

Here is example with co-ops water level stations on and search for SFBOFS - CO-OPS layer stays in the TOC and on, but no points draw: 

![image](https://user-images.githubusercontent.com/196868/62634567-226bab00-b904-11e9-944c-1f55c87ef4c8.png)

Here is an example with co-ops water level stations on and search for 'wind'  - CO-OPS layer stays in the TOC and stations with wind measurements draw:

![image](https://user-images.githubusercontent.com/196868/62634769-6fe81800-b904-11e9-9e37-9e52e90e23f9.png)

**Comment by Bobfrat** [comment]
So I stand corrected then, thanks @kknee for that detailed example.

@kbailey-noaa how do you want to proceed? Is the expected behavior to not filter the individual obs markers at all?

**Comment by kbailey-noaa** [comment]
@Bobfrat Correct re: expected behavior is to not filter the obs markers at all.  I expected that any layers I had turned on would remain on while I did a search for other layers.

While I was using it, it wasn't clear to me that I have to 'x' out the keyword I typed into the filter box in order for the markers to show back up, especially since I can turn on the filtered layer "sfbofs in this case" in the filtered view.

**Comment by cgalvarino** [comment]
To make sure I'm on the same page, we are saying that stations are all or nothing.  One casualty will be the loss of the ability to single out a station like this.

![image](https://user-images.githubusercontent.com/180985/69430689-2df0af80-0d04-11ea-8e3e-1e8cb1ad5608.png)

Is that OK?

**Comment by cgalvarino** [comment]
I'm not sure we want to go down this rabbit hole, but an alternative to the all-or-nothing approach would be to have the search apply to layers only (as @kbailey-noaa describes) and add layer-level filtering. A 2nd tier filter would allow users to drill down into something like NDBC buoys if they have a particular parameter or ID in mind.

**Comment by kbailey-noaa** [comment]
@cgalvarino Yes, all or nothing.  I actually had no idea you could type in a station ID and get results.  I personally don't have a use for that, and I suspect most IOOS Model Viewer users don't have NDBC/other station IDs memorized.  
Re: search by parameter - Doesn't the search/filter look for the keyword in the layer name, as well as the abstract? The parameter is in the layer name, so I thought that search-by-parameter functionality already exists at the 'top' level.
